internal/credit/postgresadapter/grant.go

```diff
@@ -129,6 +133,12 @@ func (g *grantDBADapter) ListActiveGrantsBetween(ctx context.Context, owner cred
 
 	grants := make([]credit.Grant, 0, len(entities))
 	for _, entity := range entities {
+		// Let's prefilter these as they are not really active, but will make the engine
+		// think they are active for that single point in time.
+		if entity.EffectiveAt.Equal(entity.ExpiresAt) {
+			continue
+		}
```

Comment:
this looks really weird to me, lets not do this (?)
What was the underlying issue?